src/modules/module_32001.c

```diff
@@ -0,0 +1,1762 @@
+  {
+    device_param->kernel_threads_max = MAX (device_param->kernel_threads_max, 1024);
+    device_param->kernel_accel_max = MAX (device_param->kernel_accel_max, 1024);
+
+    offset += snprintf (lines_buf + offset, buf_sz - offset, "%s * %u N A A\n", sanitized, hash_mode);
+    offset += snprintf (lines_buf + offset, buf_sz - offset, "CUDA * %u N A A\n", hash_mode);
+  }
+  else if ((device_param->is_opencl) && (device_param->opencl_device_vendor_id == VENDOR_ID_NV))
+  {
+    offset += snprintf (lines_buf + offset, buf_sz - offset, "# NVIDIA OpenCL uses autotune defaults (%s)\n", sanitized);
+  }
+  else if ((device_param->is_opencl) && ((device_param->opencl_device_vendor_id == VENDOR_ID_INTEL_SDK) || (device_param->opencl_device_vendor_id == VENDOR_ID_INTEL_BEIGNET)))
+  {
+    bool clamp_intel = true;
+
+    const char *env_intel_unclamp = getenv ("BIP39_ALLOW_INTEL_AUTOTUNE");
+
+    if (env_flag_is_enabled (env_intel_unclamp))
+    {
+      clamp_intel = false;
+    }
+
+    if (clamp_intel)
+    {
+      // Intel HD 630 driver (2025-10-12 harness sweep) survives autotune only when threads stay <= 32,
+      // accel <= 2, loops = 4. Clamp all Geometry knobs to the conservative configuration we run in CI.
+      device_param->kernel_threads_min = MAX (device_param->kernel_threads_min, 32);
+      device_param->kernel_threads_max = 32;
+      device_param->kernel_accel_min = MAX (device_param->kernel_accel_min, 2);
+      device_param->kernel_accel_max = 2;
+      device_param->kernel_loops_min = MAX (device_param->kernel_loops_min, 4);
+      device_param->kernel_loops_max = 4;
+
+      offset += snprintf (lines_buf + offset, buf_sz - offset, "%s * %u 32 2 4\n", sanitized, hash_mode);
+      offset += snprintf (lines_buf + offset, buf_sz - offset, "Intel * %u 32 2 4\n", hash_mode);
+    }
+    else
+    {
+      offset += snprintf (lines_buf + offset, buf_sz - offset, "# Intel OpenCL autotune unclamped for %s\n", sanitized);
+    }
+  }
+  else
+  {
+    offset += snprintf (lines_buf + offset, buf_sz - offset, "# no dynamic tuning for %s\n", sanitized);
+  }
+
+  hcfree (sanitized);
+
+  return lines_buf;
+}
+
+u32 module_kernel_threads_max (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  return KERNEL_THREADS_MAX;
+}
+
+u32 module_kernel_accel_max (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  return KERNEL_ACCEL_MAX;
+}
+
+u32 module_kernel_loops_min (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const user_options_t *user_options, MAYBE_UNUSED const user_options_extra_t *user_options_extra)
+{
+  return 1;
+}
+
+void module_hook23 (hc_device_param_t *device_param, MAYBE_UNUSED const void *hook_extra_param, MAYBE_UNUSED const void *hook_salts_buf, MAYBE_UNUSED const u32 salt_pos, const u64 pw_pos)
```

Comment:
I don't see what this hook is doing. If it's not needed, please remove it. It makes the code more complicated than it needs to be in general.